link/kprobe.go

```diff
@@ -217,8 +225,8 @@ func pmuProbe(typ probeType, symbol, path string, offset uint64, ret bool) (*per
 		return nil, fmt.Errorf("opening perf event: %w", err)
 	}
 
-	// Ensure the string pointer is not collected before PerfEventOpen returns.
-	runtime.KeepAlive(sp)
+	// Ensure PerfEventAttr is not collected before PerfEventOpen returns.
+	runtime.KeepAlive(attr)
```

Comment:
@lmb is this safe to do?

Reply:
Unfortunately not, since the garbage collector can't "see" through `Ext1`. I would return `sp` from PerfEventAttr.